src/App.tsx

```diff
@@ -15,36 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Name = 'goodName',
+  Length = 'goodLength',
+}
+
+function getVisibleGoods(
+  goods: string[],
+  { sortField, reversed }: { sortField: SortType; reversed: boolean },
+) {
+  let res = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.Name:
+        res = res.sort();
+        break;
+      case SortType.Length:
+        res = res.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (reversed) {
+    res.reverse();
+  }
+
+  return res;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const visibleGoods = getVisibleGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  const reset = () => {
+    setSortField(SortType.Default);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.Name,
+          })}
+          onClick={() => setSortField(SortType.Name)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.Length,
+          })}
+          onClick={() => setSortField(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', { 'is-light': !reversed })}
+          onClick={() => setReversed(!reversed)}
```

Comment:
Slightly better way is

```suggestion
          onClick={() => setReversed(value => !value)}
```

But no need to fix this